lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -706,9 +706,7 @@ protected Schema createSchema(Type type, Map<String, Schema> names) {
         AvroDoc annotatedDoc = c.getAnnotation(AvroDoc.class); // Docstring
         String doc = (annotatedDoc != null) ? annotatedDoc.value() : null;
         String name = c.getSimpleName();
-        String space = c.getPackage() == null ? "" : c.getPackage().getName();
-        if (c.getEnclosingClass() != null) // nested class
-          space = c.getEnclosingClass().getName().replace('$', '.');
+        String space = getNamespace(c);
```

Comment:
Would the annotation @AvroTypeName not specify the type name?

Reply:
I am not clear on this. Could you please explain this?

Do you mean to rename the function getNamespace to getTypeName ?